EngineClientTest.shouldMigrateProcessInstance is flaky

**Description**

The following test is flaky: `EngineClientTest.shouldMigrateProcessInstance`

Assertion failure:
```
Error:  Tests run: 30, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 8.865 s <<< FAILURE! -- in io.camunda.zeebe.process.test.engine.EngineClientTest
Error:  io.camunda.zeebe.process.test.engine.EngineClientTest.shouldMigrateProcessInstance -- Time elapsed: 0.450 s <<< FAILURE!
java.lang.AssertionError: 

Expecting Optional to contain a value but it was empty.
	at io.camunda.zeebe.process.test.engine.EngineClientTest.shouldMigrateProcessInstance(EngineClientTest.java:476)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

- run: https://github.com/camunda/zeebe-process-test/actions/runs/7612806113/job/20731220018

--- a/engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/process/test/engine/EngineClientTest.java
@@ -21,6 +21,7 @@ import io.camunda.zeebe.client.api.response.BrokerInfo;
 import io.camunda.zeebe.client.api.response.DeploymentEvent;
 import io.camunda.zeebe.client.api.response.EvaluateDecisionResponse;
 import io.camunda.zeebe.client.api.response.Form;
+import io.camunda.zeebe.client.api.response.MigrateProcessInstanceResponse;
 import io.camunda.zeebe.client.api.response.PartitionBrokerHealth;
 import io.camunda.zeebe.client.api.response.PartitionBrokerRole;
 import io.camunda.zeebe.client.api.response.PartitionInfo;
@@ -457,23 +458,16 @@ class EngineClientTest {
             .findFirst()
             .orElseThrow()
             .getProcessDefinitionKey();
-    zeebeClient
-        .newMigrateProcessInstanceCommand(processInstanceKey)
-        .migrationPlan(targetProcessDefinitionKey)
-        .addMappingInstruction("A", "B")
-        .send()
-        .join();
 
-    assertThat(
-            StreamSupport.stream(
-                    RecordStream.of(zeebeEngine.getRecordStreamSource())
-                        .processInstanceRecords()
-                        .spliterator(),
-                    false)
-                .filter(r -> r.getIntent() == ProcessInstanceIntent.ELEMENT_MIGRATED)
-                .filter(r -> r.getValue().getElementId().equals("B"))
-                .findFirst())
-        .isNotEmpty();
+    final MigrateProcessInstanceResponse response =
+        zeebeClient
+            .newMigrateProcessInstanceCommand(processInstanceKey)
+            .migrationPlan(targetProcessDefinitionKey)
+            .addMappingInstruction("A", "B")
+            .send()
+            .join();
+
+    assertThat(response).isNotNull();
   }
 
   @Test

